template/template.go

```diff
@@ -50,11 +51,15 @@ var (
 	})
 
 	errNaNOrInf = errors.New("value is NaN or Inf")
+
+	once sync.Once // Ensure registration happens only once
 )
 
-func init() {
-	prometheus.MustRegister(templateTextExpansionFailures)
-	prometheus.MustRegister(templateTextExpansionTotal)
+func RegisterTemplateMetrics(registry prometheus.Registerer) {
+	once.Do(func() {
```

Comment:
I think we should not use `once.Do` at all. It defeats the purpose of avoiding the global registry, which is to allow callers to register metrics with a registry of their choice. But with `once.Do`, they can do it only once.

We have to leave it to the caller to make sure they call `RegisterTemplateMetrics` only once (which is already the case, isn't it?).